frame/elections-phragmen/src/lib.rs

```diff
@@ -1113,25 +1143,33 @@ impl<T: Config> Pallet<T> {
 					// fetch deposits from the one recorded one. This will make sure that a
 					// candidate who submitted candidacy before a change to candidacy deposit will
 					// have the correct amount recorded.
-					<Members<T>>::put(
+					<Members<T>>::put::<BoundedVec<_, T::DesiredMembers>>(
 						new_members_sorted_by_id
 							.iter()
+							.take(T::DesiredMembers::get() as usize)
 							.map(|(who, stake)| SeatHolder {
 								deposit: deposit_of_candidate(who),
 								who: who.clone(),
 								stake: *stake,
 							})
-							.collect::<Vec<_>>(),
+							.collect::<Vec<_>>()
+							.try_into()
+							.expect("number members will never exceed T::DesiredMembers. qed."),
```

Comment:
I would prefer us to be defensive here, especially because this code runs on `on_initialize`. Changes in the logic/bounds may change the assumptions and `number members will never exceed T::DesiredMembers. qed.` may not hold true anymore.

I'd suggest to fail and return earlier here if the collect fails instead of using the expect.

Reply:
This code is being executed inside a closure so I cannot return the weight from here, at least not without re-organizing the code. I took the approach of taking the first `T::DesiredMembers` members so that we can be sure there will never be more members.

Is this ok?